### ISSUE TITLE
how to visualize the detection results?

### ISSUE THREAD

**Issue by HuangLian126** [YoungXIAO13/FewShotDetection#13]

**Comment by YoungXIAO13** [comment]
Hi @HuangLian126 

You can run ``test.py`` with args ``--vis``, then the visualisation is on.

**Comment by NHW2017** [comment]
> Hi @HuangLian126
> 
> You can run `test.py` with args `--vis`, then the visualisation is on.

Sorry to bother you, I also wanted to see the visual results during the test, but there was an error:
  File "/home/xxx/FewShotDetection-master/test.py", line 335, in <module>
    im = cv2.imread(imdb.image_path_from_index(int(data[4])))
  File "/home/xxx/FewShotDetection-master/lib/datasets/pascal_voc.py", line 93, in image_path_from_index
    image_path = os.path.join(self._data_path, 'JPEGImages', index + self._image_ext)
TypeError: unsupported operand type(s) for +: 'int' and 'str'
When the second error message is located, there is no problem running debug. I don’t know the reason. If you have encountered it, please let me know.

**Comment by NHW2017** [comment]
> _No description provided._

Did you successfully get the visual result?

**Comment by HuangLian126** [comment]
> > _No description provided._
> 
> Did you successfully get the visual result?



> > _No description provided._
> 
> Did you successfully get the visual result?

Yes, I successfully get the visual result. You can check your Pytorch version.

**Comment by YoungXIAO13** [comment]
Hi @NHW2017 ,

I think your problem comes from this line: 
``` image_path = os.path.join(self._data_path, 'JPEGImages', index + self._image_ext)```

Could you try modify it as:
``` image_path = os.path.join(self._data_path, 'JPEGImages', str(index) + self._image_ext)```
to see if the issue is resolved?

**Comment by NHW2017** [comment]
> Hi @NHW2017 ,
> 
> I think your problem comes from this line:
> ` image_path = os.path.join(self._data_path, 'JPEGImages', index + self._image_ext)`
> 
> Could you try modify it as:
> ` image_path = os.path.join(self._data_path, 'JPEGImages', str(index) + self._image_ext)`
> to see if the issue is resolved?

No, I have tried this modification, but a new error appears and the file path cannot be found

**Comment by NHW2017** [comment]
> > > _No description provided._
> > 
> > 
> > Did you successfully get the visual result?
> 
> > > _No description provided._
> > 
> > 
> > Did you successfully get the visual result?
> 
> Yes, I successfully get the visual result. You can check your Pytorch version.

torch==0.4.0？

**Comment by YoungXIAO13** [comment]
Then I think you might need to check the dataset is well downloaded as structured as [suggested](https://github.com/YoungXIAO13/FewShotDetection/blob/master/data/README.md#coco)

**Comment by HuangLian126** [comment]
@NHW2017 torch==0.4, torchvision==0.21

**Comment by NHW2017** [comment]
> Then I think you might need to check the dataset is well downloaded as structured as [suggested](https://github.com/YoungXIAO13/FewShotDetection/blob/master/data/README.md#coco)

I don't quite understand, because I had no problems during training and testing, but this error occurred during --vis, so I think the error is not due to the configuration of the data set

**Comment by NHW2017** [comment]
> @NHW2017 torch==0.4, torchvision==0.21

Which version of python do you use and whether you have modified other parts of the code?
